### PR TITLE
Experimental Nightly v22.1.0

### PR DESCRIPTION
**Pull request by pm-shashank-jain**
Experimental nightly based on 21.11.0 
Also changes for outstream player

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -29,7 +29,12 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
-                        return controllerPaths[conf.pwt.adserver || "DFP"];
+			if (conf.pwt.identityOnly == "1") {
+				console.log("Returning controller path as -> IDHUB");
+				return controllerPaths["IDHUB"];
+			}
+                        console.log("Returning controller path as -> "+conf.pwt.adserver);
+                        return controllerPaths[conf.pwt.adserver || 'DFP']
```
Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

**Comment by sureshghare** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- src_new/util.js  2
- webpack.config.js  1
- src_new/controllers/idhub.js  2
         

Clones added
============
- src_new/adapters/prebid.js  2
- src_new/controllers/idhub.js  2
         
```


See the complete overview on [Codacy](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204&bid=49906)

File: webpack.config.js [hunk 2]
```diff
@@ -29,7 +29,12 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
-                        return controllerPaths[conf.pwt.adserver || "DFP"];
+			if (conf.pwt.identityOnly == "1") {
+				console.log("Returning controller path as -> IDHUB");
+				return controllerPaths["IDHUB"];
+			}
+                        console.log("Returning controller path as -> "+conf.pwt.adserver);
```
Comment (sureshghare) [inline webpack.config.js:36]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 3]
```diff
@@ -29,7 +29,12 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
-                        return controllerPaths[conf.pwt.adserver || "DFP"];
+			if (conf.pwt.identityOnly == "1") {
+				console.log("Returning controller path as -> IDHUB");
```
Comment (sureshghare) [inline webpack.config.js:33]: Codacy Issue found: [Expected indentation of 9 tabs but found 28 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 1 above)
Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Missing semicolon.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 4]
```diff
@@ -29,7 +29,12 @@ module.exports = {
...
+				return controllerPaths["IDHUB"];
```
Comment (sureshghare) [inline webpack.config.js:34]: Codacy Issue found: [Expected indentation of 9 tabs but found 4.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 1 above)
Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Strings must use doublequote.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

**Comment by sureshghare** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- webpack.config.js  1
- src_new/controllers/idhub.js  2
         

Clones added
============
- src_new/adapters/prebid.js  2
- src_new/controllers/idhub.js  2
         
```


See the complete overview on [Codacy](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204&bid=49906)

File: webpack.config.js (hunk 3 above)
Comment (sureshghare) [inline webpack.config.js:33]: Codacy Issue found: [Expected indentation of 9 tabs but found 28 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 2 above)
Comment (sureshghare) [inline webpack.config.js:36]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 4 above)
Comment (sureshghare) [inline webpack.config.js:34]: Codacy Issue found: [Expected indentation of 9 tabs but found 4.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 1 above)
Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Missing semicolon.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

Comment (sureshghare) [inline webpack.config.js:37]: Codacy Issue found: [Strings must use doublequote.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

**Comment by pm-prakhar-jain** [comment]
Hi Shashank, 

Please look into codacy issues reported for this PR once to check if any of these is alarming.

Thanks,
Prakhar Jain

**Comment by sureshghare** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- webpack.config.js  1
- src_new/controllers/idhub.js  2
         

Clones added
============
- src_new/adapters/prebid.js  2
- src_new/controllers/idhub.js  2
         
```


See the complete overview on [Codacy](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204&bid=49906)

File: webpack.config.js [hunk 5]
```diff
@@ -29,6 +29,11 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
+                        if (conf.pwt.identityOnly == "1") {
+                          console.log("Returning controller path as -> IDHUB");
```
Comment (sureshghare) [inline webpack.config.js:33]: Codacy Issue found: [Expected indentation of 9 tabs but found 28 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 6]
```diff
@@ -29,6 +29,11 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
+                        if (conf.pwt.identityOnly == "1") {
```
Comment (sureshghare) [inline webpack.config.js:32]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 7]
```diff
@@ -29,6 +29,11 @@ module.exports = {
...
+                          console.log("Returning controller path as -> IDHUB");
+                          return controllerPaths["IDHUB"];
+                        }
+                        console.log("Returning controller path as -> "+conf.pwt.adserver);
```
Comment (sureshghare) [inline webpack.config.js:36]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 8]
```diff
@@ -29,6 +29,11 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
+                        if (conf.pwt.identityOnly == "1") {
+                          console.log("Returning controller path as -> IDHUB");
+                          return controllerPaths["IDHUB"];
```
Comment (sureshghare) [inline webpack.config.js:34]: Codacy Issue found: [Expected indentation of 9 tabs but found 4.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 6 above)
Comment (pm-manasi-moghe) [reply webpack.config.js:32]: done

File: webpack.config.js (hunk 7 above)
Comment (pm-manasi-moghe) [reply webpack.config.js:36]: done

File: webpack.config.js (hunk 8 above)
Comment (pm-manasi-moghe) [reply webpack.config.js]: done

**Comment by sureshghare** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src_new/controllers/idhub.js  2
- webpack.config.js  1
         

Clones added
============
- src_new/controllers/idhub.js  2
- src_new/adapters/prebid.js  2
         
```


See the complete overview on [Codacy](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204&bid=49906)

File: webpack.config.js [hunk 9]
```diff
@@ -29,6 +29,11 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
+                        if (conf.pwt.identityOnly == "1") {
+                            console.log("Returning controller path as -> IDHUB");
```
Comment (sureshghare) [inline webpack.config.js:33]: Codacy Issue found: [Expected indentation of 9 tabs but found 28 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 6 above)
Comment (sureshghare) [inline webpack.config.js:32]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 10]
```diff
@@ -29,6 +29,11 @@ module.exports = {
...
+                            return controllerPaths["IDHUB"];
+                        }
+                        console.log("Returning controller path as -> "+conf.pwt.adserver);
```
Comment (sureshghare) [inline webpack.config.js:36]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 11]
```diff
@@ -29,6 +29,11 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
+                        if (conf.pwt.identityOnly == "1") {
+                            console.log("Returning controller path as -> IDHUB");
+                            return controllerPaths["IDHUB"];
```
Comment (sureshghare) [inline webpack.config.js:34]: Codacy Issue found: [Expected indentation of 9 tabs but found 4.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

**Comment by sureshghare** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- webpack.config.js  1
- src_new/controllers/idhub.js  2
         

Clones added
============
- src_new/adapters/prebid.js  2
- src_new/controllers/idhub.js  2
         
```


See the complete overview on [Codacy](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204&bid=49906)

File: webpack.config.js (hunk 3 above)
Comment (sureshghare) [inline webpack.config.js:33]: Codacy Issue found: [Expected indentation of 9 tabs but found 28 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 12]
```diff
@@ -29,7 +29,12 @@ module.exports = {
                     {
                       pattern: /%%PATH_TO_CONTROLLER%%/g,
                       replacement: function (match, p1, offset, string) {
-                        return controllerPaths[conf.pwt.adserver || "DFP"];
+			if (conf.pwt.identityOnly == "1") {
```
Comment (sureshghare) [inline webpack.config.js:32]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js [hunk 13]
```diff
@@ -29,7 +29,12 @@ module.exports = {
...
+				console.log("Returning controller path as -> IDHUB");
+				return controllerPaths["IDHUB"];
+			}
+			console.log("Returning controller path as -> "+conf.pwt.adserver);
```
Comment (sureshghare) [inline webpack.config.js:36]: Codacy Issue found: [Expected indentation of 8 tabs but found 24 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

File: webpack.config.js (hunk 4 above)
Comment (sureshghare) [inline webpack.config.js:34]: Codacy Issue found: [Expected indentation of 9 tabs but found 4.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15204)

**Pull request merged**